### ISSUE TITLE
Selenium and Kotlintest

### ISSUE THREAD

**Issue by michaeloa** [kotlintest/kotlintest#235]
Hi.
I'm messing around with Selenium and Kotlintest. Quite easy to get this starting to work, but I'm running into an issue. I want to be able to start up the browser, do some stuff on a page, and then run a bunch of tests on the page. E.g., (using WordSpec, my preferred format):
```
RunSearchTest : WordSpec() {
  init {
  "When I run search the page" should {
    // ***
    // start up chrome driver, initialize page, do the search
    // ***
    "contain exactly 1 result" { 
      // Test
    }
   "contain a search again button" {
      // Test
    }
  }
}
```
This works, but not exactly the way I want it to, of course - it initiates 3? webpages, and runs the test on the last two. I only want to initiate the selenium webdriver and navigate the pages once for each set of tests, of course, since it's so expensive.

Interceptors do not seem to be the answers to this problem. Wondering if there is an obvious simple solution to this that I've overlooked.

**Comment by Kerooker** [comment]
(Sorry for the very late reply, I'm looking at older issues now)

Due to how the isolation works, for every test the whole spec is executed. So `"When I run search the page`" will be executed every time before the test scenarios.

If this isn't the behavior you want, could you explain a little more what you're looking for?

**Comment by Kerooker** [comment]
For instance, the following behavior works:

```
class Foo(var int: Int = 0)
    
    init {
        val foo = Foo()
    
        "Foo" should {
            foo.int++
            
            "Should be executed only once" {
                foo.int shouldBe 1
            }
            
            "Not twice" {
                foo.int shouldBe 1
            }
            
            "Not thrice" {
                foo.int shouldBe 1
            }
        }
    }
```

`foo.int++` is executed only once, and not a lot of times. So your selenium browser should open only once! Tell me if I didn't understand this correctly

**Comment by michaeloa** [comment]
Thanks for the response, though we've moved on from this as kotlintest (at the time at least - more than a year ago), didn't meet all our requirements.

The way the tests ran, it certainly seemed like foo.int++ was executed multiple times - or at least Selenium/ChromeDriver was, in the way it interacted with kotlintest. Might be some weirdness in the latter too, though - we had some issues with the version we were using even when working with plain junit before we found a good solution.

The test code involved here is lost, though (it was just a prototype), so I can't go back to check it out.
